### PR TITLE
[WIP] use macros to define macros for matchers

### PR DESCRIPTION
**Pull request by jbcoe**

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -687,32 +680,28 @@ mod tests {
...
-        matcher: LetStatementMatcher{
-            identifier: "x".to_string(),
-            mutable: false,
-            matcher: Box::new(BinaryOperatorExpressionMatcher{
-                left: Box::new(IdentifierMatcher{identifier: "a".to_string()}),
-                right: Box::new(IdentifierMatcher{identifier: "b".to_string()}),
-                operator: OpName::Divide,
-            })
-        },
+        matcher: match_let_stmt!(
```
Comment (philipcraig) [inline src/parser.rs]: why abbreviate to `stmt` here and throughout?

**Review by philipcraig — CHANGES_REQUESTED** [review]
I like how both the macros and the tests written using the macros have been shorted in this PR

File: src/parser.rs (hunk 1 above)
Comment (jbcoe) [reply src/parser.rs]: Well-spotted it's inconsistent and needs fixing.

Comment (jbcoe) [reply src/parser.rs]: Fixed.

**Pull request merged**